docs/manual/shared_memory.rst

```diff
@@ -60,78 +136,181 @@ Save the following xml as cyclonedds.xml.
 
-***********
-Performance
-***********
+* Liveliness
 
-A performance improvement can be observed for sufficiently large sample sizes and data transfer on the same machine (otherwise the regular network transmission is used and their will be no performance gain).
+  * DDS_LIVELINESS_AUTOMATIC
 
-Once the loan interface is fully implemented transmission speed will be independent of the sample size when shared memory is used, i.e. take constant time.
+* Deadline
 
-*************
-To developers
-*************
+  * DDS_INFINITY
+
+* Reliability
+
+  * DDS_RELIABILITY_RELIABLE
+
+* Durability
+
+  * DDS_DURABILITY_VOLATILE
+
+* History
+
+  * DDS_HISTORY_KEEP_LAST
+
+  * with depth no larger than the subscriber history request as set in the configuration file
+
+If any of these prerequisites are not satisfied, shared memory exchange will be disabled and data transfer will fall back to the network interface.
+
+A further limitation is the maximum number of subscriptions per process for the iceoryx service, which is 127.
+
+Lastly, there is the limit on the operating system. Iceoryx currently has no functioning implementation for the Windows operating system, this is `under development <https://github.com/eclipse/iceoryx/issues/33>`_.
+
+Loan Mechanism
+--------------
+
+If the choice to use shared memory exchange is made, additional performance gains can be made by using the loan mechanism on the writer side.
+The loan mechanism directly allocates memory from the iceoryx shared memory pool, and provides this to the user in the shape of the message data type.
+Thereby eliminating a copy step in the publication process.
+
+.. code-block:: C
+
+  message_type *loaned_sample;
+  dds_return_t status = dds_loan_sample(writer, (void**)&loaned_sample);
+
+If *status* returns **DDS_RETCODE_OK**, then *loaned_sample* will contain a pointer to the memory pool object, in all other cases, *loaned_sample* should not be dereferenced.
+For requesting loaned samples, the writer used to request the loaned sample should be of the same data type as the sample that you are writing in it, since necessary information about the data type is supplied by the writer.
+
+The user is limited in this case by the maximum number of outstanding loans, defined by **MAX_PUB_LOANS** (default set to 8). This is the maximum number of loaned samples that can be outstanding process wide from the shared memory, before some must be returned (handed back to the publisher through *dds_write*) before requesting new loaned samples.
+
+After a loaned sample has been returned to the shared memory pool (at the moment, this can only be done by invoking *dds_write*), dereferencing the pointer is undefined behaviour.
+
+If the user is not able to use the loan mechanism, as *dds_write* will also write to the shared memory service if it is configured to do so. Though in that case, the overhead of the additional copy step in publication is still incurred.
```

Comment:
typo: a *dds_write*
Maybe explain the copy step, i.e. that in this case write loans internally and performs a copy into loaned memory. May be obvious though.

Reply:
The asterisks on both sides of dds_write are .rst style markers, causing the text to show up as italics in the generated documentation.

About the extra copy step, where do we draw the line between documenting necessary details and trivialities? In this case, I think it will cause no harm going into a little more detail.

Reply:
I think the explanation level of detail is fine. This is always subjective but I tend more to the "explaining more avoids future questions/bug reports" side. :-)